test_runner/fixtures/neon_fixtures.py

```diff
@@ -4996,13 +4996,39 @@ def check_restored_datadir_content(
     assert (mismatch, error) == ([], [])
 
 
+# wait for subscriber to catch up with publisher
 def logical_replication_sync(
     subscriber: PgProtocol,
     publisher: PgProtocol,
+    # pass subname explicitly to avoid confusion
+    # when multiple subscriptions are present
+    subname: str,
     sub_dbname: str | None = None,
     pub_dbname: str | None = None,
-) -> Lsn:
+):
     """Wait logical replication subscriber to sync with publisher."""
+    not_ready = True
+    while not_ready:
+        # first check if the subscription is active `r` = `ready` normal state
+        query = f"""SELECT 1 FROM pg_subscription_rel join pg_catalog.pg_subscription
+                    on pg_subscription_rel.srsubid = pg_subscription.oid
+                    WHERE srsubstate NOT IN ('r', 's') and subname='{subname}'"""
+
+        if sub_dbname is not None:
+            res = subscriber.safe_psql(query, dbname=sub_dbname)
+        else:
+            res = subscriber.safe_psql(query)
+
+        log.info(f"res: {res}")
+        if res is None:
+            not_ready = False
+        if len(res) == 0:
+            not_ready = False
+
+        time.sleep(0.5)
```

Comment:
We have a `wait_until` utility that might be useful for you to use here.

Reply:
good idea. updated the code